### PR TITLE
Remove __repr__ method in DummyNode

### PR DESCRIPTION
**Pull request by atombrella**
There are no tests for either DummyNode nor Node. I suppose this would be in `test_migrations`, perhaps as a new file?

### CODE REVIEW

**Comment by timgraham** [comment]
I think tests could go in `test_graph.py`.

**Comment by atombrella** [comment]
OK. The `%r,  %r` in `repr` didn't work for me, as the key always seems to be a tuple with two elements.

I added a test for promote as well.

**Comment by timgraham** [comment]
If no functionality is changed here, it would be clearest to separate the tests (adding test coverage) and code changes (cosmetic cleanups) into separate commits.

File: tests/migrations/test_graph.py [hunk 1]
```diff
@@ -407,3 +409,21 @@ def test_stringify(self):
 
         self.assertEqual(str(graph), "Graph: 5 nodes, 3 edges")
         self.assertEqual(repr(graph), "<MigrationGraph: nodes=5, edges=3>")
+
+
+class NodeTest(SimpleTestCase):
```
Comment (timgraham) [inline tests/migrations/test_graph.py]: Tests

File: tests/migrations/test_graph.py [hunk 2]
```diff
@@ -407,3 +409,21 @@ def test_stringify(self):
...
+    def test_node_repr(self):
+        node = Node(("app_a", "0001"))
```
Comment (timgraham) [inline tests/migrations/test_graph.py]: Use single quotes unless the string contains single quotes.

File: tests/migrations/test_graph.py [hunk 3]
```diff
@@ -407,3 +409,21 @@ def test_stringify(self):
...
+        self.assertEqual(repr(node), "<Node: ('app_a', '0001')>")
+
+    def test_dummynode_repr(self):
+        node = DummyNode(("app_a", "0001"), origin="app_a.0001", error_message='x is missing')
+        self.assertEqual(repr(node), "<DummyNode: ('app_a', '0001')>")
+
+    def test_dummynode_promote(self):
+        dummy = DummyNode(key=("app_a", "0001"), origin="app_a.0002",
```
Comment (timgraham) [inline tests/migrations/test_graph.py]: Use hanging indent:
````
node = DummyNode(
    ...,
    ...,
)
```

**Comment by atombrella** [comment]
OK. Tests are actually also missing for the `ExtendsNode`\'s `repr`-method.

**Comment by atombrella** [comment]
I can add them, but I don't spot a good place. The template test suite is fairly large, and the functionality of `extends` is already tested.

**Comment by timgraham** [comment]
Absent another obvious location, I guess I'd expect the tests in `tests/template_tests/syntax_tests/test_extends.py`.

**Comment by atombrella** [comment]
@timgraham Can you please check the last test for `ExtendsNode` if this is something that seems ok? It's a bit tricky to see what the input arguments are supposed to be, so I browsed for something with `token`.

**Comment by atombrella** [comment]
@timgraham I found a few more cases, but in the pattern `<IfNode>`. I suppose this #27546 could be used as a reference for the commit messages?

**Pull request merged**